src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4122,315 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
 #endif // DEBUG
 }
 
+/*****************************************************************************
+ *  Checks all jumps in the jump list to see if they are unconditional jumps
+ * generated at the end of BB_ALWAYS basic blocks and whether they are now a jump
+ * to the next instruction, this can happen when the following basic block emitted
+ * no instructions
+ *
+ * returns true if any jumps have been removed, otherwise returns false
+ */
+bool emitter::emitRemoveJumpToNextInst()
+{
+#ifdef TARGET_XARCH
+#ifdef DEBUG
+    if (emitComp->verbose)
+    {
+        printf("*************** In emitRemoveJumpToNextInst()\n");
+    }
+    if (EMIT_INSTLIST_VERBOSE)
+    {
+        printf("\nInstruction list before unconditional jump to next instruction removal:\n\n");
+        emitDispIGlist(true);
+    }
+    if (EMITVERBOSE)
+    {
+        emitDispJumpList();
+    }
+#endif
+
+    instrDescJmp*  jmp;
+    UNATIVE_OFFSET adjIG;
+    UNATIVE_OFFSET adjLJ;
+    insGroup*      lstIG;
+    int            removedCount   = 0;
+    UNATIVE_OFFSET removedSize    = 0;
+    int            iterationCount = 1;
+    bool           jmpRemoved     = false;
+
+/*****************************************************************************/
+/* If we iterate to look for more jumps to remove, we start again here.     */
+/*****************************************************************************/
+
+AGAIN:
+
+    lstIG                     = nullptr;
+    adjLJ                     = 0;
+    adjIG                     = 0;
+    jmp                       = emitJumpList;
+    instrDescJmp* previousJmp = nullptr;
+
+    while (jmp)
+    {
+        // if the jump is unconditional then it is a candidate
+        if (jmp->idInsFmt() == IF_LABEL && emitIsUncondJump(jmp) && !jmp->idjKeepLong && jmp->idGetJmpAlwaysFlag())
+        {
+            // unset the flag so that we know it has been processed already
+            jmp->idSetJmpAlwaysFlag(0);
+
+            // target group is not bound yet so use the cookie to fetch it
+            insGroup* targetGroup = (insGroup*)emitCodeGetCookie(jmp->idAddr()->iiaBBlabel);
+
+            if (targetGroup != nullptr && jmp->idjIG->igNext == targetGroup)
```

Comment:
Why is this not like this?

```suggestion
            if (targetGroup != nullptr && jmp->idjIG->igNum + 1 == targetGroup->igNum)
```

Reply:
ah, never mind I misread the code.

Reply:
Because that's more complex than just comparing the pointers directly. Is there a specific preference for using the `igNum`?